#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("remove_ramp", bool_t, 6, "Whether to apply ramp removal", False)
gen.add("cut_off_distance", double_t, 7, "Distance at which readings get cut off to avoid noise / wrong fitting", 1.2, 0.0, 10.0)
gen.add("plane_fitting_type", int_t, 8, "Algorithm used to select plane, see pcl sample consensus method_types header and docu", 1, 0, 6)
gen.add("plane_max_search_count", int_t, 9, "Max number of search circles to detect planes", 40, 1, 100000)
gen.add("plane_max_deviation", double_t, 10, "Max. deviation for points to be considered inlier of plane", 0.025, 0.0, 1.0)
gen.add("ground_plane_padding_size", double_t, 11, "Size [m] of padding to check if ramp is connected to ground plane", 0.02, 0.0, 1.0)
gen.add("plane_max_degree", double_t, 12, "Maximum angular deviation of planes (ground and ramp) in deg", 8.0, 0.0, 360.0)

gen.add("normal_neighbours", int_t, 13, "Number of neighbour points for normal calulcation", 8, 1, 1000)
gen.add("normal_distance_weight", double_t, 14, "Distance normal weight for ground plane, see pcl docu", 0.025, 0.0, 1.0)
gen.add("normal_distance_weight_ramp", double_t, 15, "Distance normal weight for ramp plane, see pcl docu", 0.0001, 0.0, 1.0)

exit(gen.generate("downsampler", "downsampler", "DownsamplerRamp"))
